chcemvediet/apps/inforequests/tests/test_models/test_deadline.py

```diff
@@ -0,0 +1,157 @@
+        self.assertEqual(deadline.workdays_passed_at(naive_date(u'2010-10-10')), 3)
+
+    def test_calendar_days_remaining_property_and_calendar_days_remaining_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-08'))
+        self.assertEqual(deadline.calendar_days_remaining, 5)
+        self.assertEqual(deadline.calendar_days_remaining_at(naive_date(u'2010-10-08')), 5)
+
+    def test_workdays_remaining_property_and_workdays_remaining_at_method(self):
+        deadline =  self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-08'))
+        self.assertEqual(deadline.workdays_remaining, 3)
+        self.assertEqual(deadline.workdays_remaining_at(naive_date(u'2010-10-08')), 3)
+
+    def test_calendar_days_behind_property_and_calendar_days_behind_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-26'))
+        self.assertEqual(deadline.calendar_days_behind, 13)
+        self.assertEqual(deadline.calendar_days_behind_at(naive_date(u'2010-10-26')), 13)
+
+    def test_workdays_behind_property_and_workdays_behind_at_method(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, _today=naive_date(u'2010-10-26'))
+        self.assertEqual(deadline.workdays_behind, 9)
+        self.assertEqual(deadline.workdays_behind_at(naive_date(u'2010-10-26')), 9)
+
+    def test_is_deadline_missed_property_and_is_deadline_missed_at_method(self):
+        tests = (                            # _today,                    is_deadline_missed
+                (Deadline.UNITS.CALENDAR_DAYS, naive_date(u'2010-10-13'), False),
+                (Deadline.UNITS.CALENDAR_DAYS, naive_date(u'2010-10-14'), True),
+                (Deadline.UNITS.WORKDAYS,      naive_date(u'2010-10-15'), False),
+                (Deadline.UNITS.WORKDAYS,      naive_date(u'2010-10-16'), True),
+        )
+        for unit, _today, is_deadline_missed in tests:
+            deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=unit, _today=_today)
+            self.assertEqual(deadline.is_deadline_missed, is_deadline_missed)
+            self.assertEqual(deadline.is_deadline_missed_at(_today), is_deadline_missed)
+
+    def test_snooze_date_property(self):
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-14'))
+        self.assertEqual(deadline.snooze_date, naive_date(u'2010-10-14'))
+        deadline = self._create_deadline(base_date=naive_date(u'2010-10-05'), value=8, unit=Deadline.UNITS.CALENDAR_DAYS, snooze=naive_date(u'2010-10-12'))
```

Comment:
@viliambalaz Pls v ramci jedneho testu nerecykluj premenne. Ked su 2x `deadline` pouzi `deadline1` a `deadline2`. Je to prehladnejsie, ked sa da spolahnut na to, ze ked raz nieco do premennej priradis, tak to v nej ostane.

Reply:
:ok: